Remotion/Mixins/UnitTests/Core/CodeGeneration/IntegrationTests/MixedTypeCodeGeneration/InstantiationTest.cs

```diff
@@ -43,9 +43,13 @@ public void GeneratedTypeCanBeInstantiatedViaCtorCall ()
     public void GeneratedTypeCanBeInstantiatedViaCtorCallEvenWhenDerived ()
     {
       Type generatedType = TypeFactory.GetConcreteType (typeof (BaseType3));
-
+#if NETFRAMEWORK
       AssemblyBuilder builder =
           AppDomain.CurrentDomain.DefineDynamicAssembly (new AssemblyName ("Foo"), AssemblyBuilderAccess.Run);
+#else
+      AssemblyBuilder builder =
+          AssemblyBuilder.DefineDynamicAssembly (new AssemblyName ("Foo"), AssemblyBuilderAccess.Run);
+#endif
```

Comment:
no switch needed here, just use 
```suggestion
      AssemblyBuilder builder =
          AssemblyBuilder.DefineDynamicAssembly (new AssemblyName ("Foo"), AssemblyBuilderAccess.Run);
```